src/main/webapp/WEB-INF/view/activityfeed.jsp

```diff
@@ -22,6 +31,30 @@
     <h1>Activity</h1>
 
     <p>Here's everything that's happened on the site so far!</p>
+
+    <div id="activity">
+      
+      <%List<User> users = 
+        (List<User>) request.getAttribute("users");
+      if (users != null && !users.isEmpty()) {
```

Comment:
I think it's not needed since the for each loop simply won't execute if it's empty and I don't think users will be null. I'll remove it. Sorry, I'm still used to all the memory stuff and pointers from C++.

Reply:
Yup, no worries.